workflow/controller/workflowpod.go

```diff
@@ -815,6 +826,28 @@ func addSchedulingConstraints(pod *apiv1.Pod, wfSpec *wfv1.WorkflowSpec, tmpl *w
 	}
 }
 
+// GetBoundaryTemplate get a template through the node's BoundaryID.
+func (woc *wfOperationCtx) GetBoundaryTemplate(nodeName string) (*wfv1.Template, error) {
```

Comment:
Can something like this function also be used from https://github.com/argoproj/argo-workflows/blob/e00abd1ded9e8f4787093b84f6cb26ee0870ec7c/workflow/controller/operator.go#L2740

The two bits of code are very similar to each other, and it's good to extract common functionality whilst we're doing this kind of work.

This function could take a *NodeStatus instead

Reply:
OK， I will try it.

Reply:
I have made some reuses, please take a look，Thanks.